.gitlab-ci.yml

```
@@ -18,3 +24,32 @@ test:unit:
     - git submodule update --init --recursive
   script:
     - make test
+
+test:integration:
+  stage: test
+  needs: []
+  image: docker:18-dind
+  tags:
+    - mender-qa-slave
+  before_script:
+    - /usr/local/bin/dockerd-entrypoint.sh &
+    - sleep 10
+    - export DOCKER_HOST="unix:///var/run/docker.sock"
+    - docker version
+    - export
+    # Install test dependencies
+    - apk --update --no-cache add bash gcc make curl util-linux openssh
+      python3 python3-dev libc-dev libffi-dev openssl-dev libc6-compat git
+    - CRYPTOGRAPHY_DONT_BUILD_RUST=1 pip3 install -r tests/integration/requirements.txt
+    - pip3 install docker-compose==1.24.0
```

To keep this in sync with the submodule, I suggest something like:
```suggestion
    - apk --update --no-cache $(cat tests/integration/mender_integration/tests/requirements/apk-requirements.txt)
    - pip install -r tests/integration/mender_integration/tests/requirements/python-requirements.txt
```

And update the submodule to today's master.

This will allow us to update the integration tests framework later, and make sure that things are aligned

Some notes:
* There might be missing OS dependencies (git?)
* docker-compose installation in regular integ tests was hacky at the time you started working on this, but it was fixed at https://github.com/mendersoftware/mender-qa/pull/480
* The `CRYPTOGRAPHY_DONT_BUILD_RUST=1` won't be needed as `apk-requirements.txt` added rust the moment that cryptography was updated: https://github.com/mendersoftware/integration/pull/1197
* If tests in this repo has extra dependencies, only these should be added to a local `tests/integration/requirements.txt`

What do you think overall of this approach?

---

looks great, I've reworked it this way